# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1319774030 25200
#      Thu Oct 27 20:53:50 2011 -0700
# Node ID 4539977175982bb2b1a0a85c9cda6e507bb379ff
# Parent  0d4c6d54b6a3240a2dab58f5dbdf8695e865456c
Add functional tests for csvwrite/csvread.

* Add functional tests for csvwrite/csvread.

diff --git a/scripts/io/csvread.m b/scripts/io/csvread.m
--- a/scripts/io/csvread.m
+++ b/scripts/io/csvread.m
@@ -28,8 +28,14 @@
 ## @end example
 ##
 ## @seealso{csvwrite, dlmread, dlmwrite}
 ## @end deftypefn
 
 function x = csvread (filename, varargin)
   x = dlmread (filename, ",", varargin{:});
 endfunction
+
+
+%% Tests for csvread() are in csvwrite()
+%% Mark file as being tested
+%!assert (1)
+
diff --git a/scripts/io/csvwrite.m b/scripts/io/csvwrite.m
--- a/scripts/io/csvwrite.m
+++ b/scripts/io/csvwrite.m
@@ -29,8 +29,26 @@
 ## @end example
 ##
 ## @seealso{csvread, dlmwrite, dlmread}
 ## @end deftypefn
 
 function csvwrite (filename, x, varargin)
   dlmwrite (filename, x, ",", varargin{:});
 endfunction
+
+
+%!shared fname
+%! fname = tmpnam ();
+
+%!test
+%! csvwrite (fname, magic (3));
+%! assert (csvread (fname), magic (3));
+%! unlink (fname);
+
+%!test
+%! csvwrite (fname, magic (3), "precision", "%2.1f", "newline", "unix");
+%! fid = fopen (fname, "rt");
+%! txt = char (fread (fid,Inf,'char')');
+%! fclose (fid);
+%! assert (txt, "8.0,1.0,6.0\n3.0,5.0,7.0\n4.0,9.0,2.0\n");
+%! unlink (fname);
+
